pkg/controller/jobs/appwrapper/appwrapper_controller.go

```diff
@@ -122,6 +122,9 @@ func (aw *AppWrapper) PodSets() ([]kueue.PodSet, error) {
 		ctrl.Log.Error(err, "Error returned from awutils.GetPodSets", "appwrapper", aw)
 		return nil, err
 	}
+	for idx := range podSets {
+		podSets[idx].TopologyRequest = jobframework.PodSetTopologyRequest(&podSets[idx].Template.ObjectMeta, nil, nil, nil)
```

Comment:
Right, there is no rank-based ordering for now. 

IIUC this requires checking the type of the wrapped job and preparing configuration depending on this. 
We could probably register some "rank-based ordering" factory by framework, and you could do the lookup here to construct the input.

However, this is substantially more work so I suggest it as a follow up.